interfaces/builtin/ros_opt_data.go

```diff
@@ -0,0 +1,65 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const rosOptDataSummary = `Allows read-only access to the static data such as xacro,yaml,urdf,stl,... in the standard /opt/ros folder`
+
+const rosOptDataBaseDeclarationSlots = `
+  ros-opt-data:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: false
+`
+
+const rosOptDataConnectedPlugAppArmor = `
+# Description: Allows read-only access to the xacro,yaml,urdf,stl files in /opt/ros folder
+# capability dac_read_search,  # this should not be necessary, the idea is to read only what normal user can read
+/var/lib/snapd/hostfs/opt/ros/ r,
+/var/lib/snapd/hostfs/opt/ros/**/ r,
+/var/lib/snapd/hostfs/opt/ros/**/*.config r,        # for robot configuration
```

Comment:
why not just allow read only access to everything under `/opt/ros/`? why only restrict it to these particular file extensions?

Reply:
Just to be clear from the beginning, I know filtering by extensions is not related to security since any file can be renamed in the host to match the filter and be accessible from the snap.
The reason I filtered is to clearly define what should be used and what not. In particular *.so files and executables are not going to be compatible by default unless the host uses the same Ros version as the snap. 
While we want this interface to be used for common file formats related to robotics data. Data files are usually compatible with multiple versions of Ros/Ubuntu and are not well suited to be distributed inside application snaps, so it's convenient to keep them in the host and use them in the snaps.

Reply:
Should the interface perhaps be called something like `ros-host-data` then instead?

Reply:
@alexmurray  no strong opinion on naming, I think I followed an advice from a meeting with Michael Vogt but it was many months ago. 
Isn't "host" implied by the concept of interface itself (interfacing snaps with the host system)? 
With "opt" I am telling users they cannot put their ros data in a random place in their host and expect it to be visible inside the snap.

Reply:
we never use "host" interface names, except hostname-control, usually in the few cases that we want to point out about the host we use "system" like in system-packages-doc,  if we follow that convention it would become "system-ros-opt-data" otherwise ros-opt-data is also fine I think